set exit code when go test "setup failed"

When go has a "setup failed", using `-set-exit-code` still returns a 0 exit code.
Related to #46

example for test output:
```
# github.com/name/project/package
packge/file.go:15:2: cannot find package "other/package" in any of:
	/path/venodr (vendor tree)
	/path/go/root (from $GOROOT)
	/path/go/path (from $GOPATH)
FAIL	github.com/name/project/package [setup failed]
```

--- a/go-junit-report_test.go
+++ b/go-junit-report_test.go
@@ -520,7 +520,7 @@ var testCases = []TestCase{
 					Name: "package/name/failing1",
 					Tests: []*parser.Test{
 						{
-							Name:   "build failed",
+							Name:   "[build failed]",
 							Time:   0,
 							Result: parser.FAIL,
 							Output: []string{
@@ -533,11 +533,27 @@ var testCases = []TestCase{
 					Name: "package/name/failing2",
 					Tests: []*parser.Test{
 						{
-							Name:   "build failed",
+							Name:   "[build failed]",
 							Time:   0,
 							Result: parser.FAIL,
 							Output: []string{
 								"failing2/another_failing_test.go:20: undefined: y",
+							},
+						},
+					},
+				},
+				{
+					Name: "package/name/setupfailing1",
+					Tests: []*parser.Test{
+						{
+							Name:   "[setup failed]",
+							Time:   0,
+							Result: parser.FAIL,
+							Output: []string{
+								"setupfailing1/failing_test.go:4: cannot find package \"other/package\" in any of:",
+								"\t/path/vendor (vendor tree)",
+								"\t/path/go/root (from $GOROOT)",
+								"\t/path/go/path (from $GOPATH)",
 							},
 						},
 					},

--- a/parser/parser.go
+++ b/parser/parser.go
@@ -42,7 +42,7 @@ type Test struct {
 var (
 	regexStatus   = regexp.MustCompile(`^\s*--- (PASS|FAIL|SKIP): (.+) \((\d+\.\d+)(?: seconds|s)\)$`)
 	regexCoverage = regexp.MustCompile(`^coverage:\s+(\d+\.\d+)%\s+of\s+statements$`)
-	regexResult   = regexp.MustCompile(`^(ok|FAIL)\s+([^ ]+)\s+(?:(\d+\.\d+)s|(\[build failed]))(?:\s+coverage:\s+(\d+\.\d+)%\sof\sstatements)?$`)
+	regexResult   = regexp.MustCompile(`^(ok|FAIL)\s+([^ ]+)\s+(?:(\d+\.\d+)s|(\[\w+ failed]))(?:\s+coverage:\s+(\d+\.\d+)%\sof\sstatements)?$`)
 	regexOutput   = regexp.MustCompile(`(    )*\t(.*)`)
 )
 
@@ -98,11 +98,11 @@ func Parse(r io.Reader, pkgName string) (*Report, error) {
 			if matches[5] != "" {
 				coveragePct = matches[5]
 			}
-			if matches[4] == "[build failed]" {
+			if strings.HasSuffix(matches[4], "failed]") {
 				// the build of the package failed, inject a dummy test into the package
 				// which indicate about the failure and contain the failure description.
 				tests = append(tests, &Test{
-					Name:   "build failed",
+					Name:   matches[4],
 					Result: FAIL,
 					Output: packageCaptures[matches[2]],
 				})
@@ -142,7 +142,7 @@ func Parse(r io.Reader, pkgName string) (*Report, error) {
 			testsTime += testTime
 		} else if matches := regexCoverage.FindStringSubmatch(line); len(matches) == 2 {
 			coveragePct = matches[1]
-		} else if matches := regexOutput.FindStringSubmatch(line); len(matches) == 3 {
+		} else if matches := regexOutput.FindStringSubmatch(line); capturedPackage == "" && len(matches) == 3 {
 			// Sub-tests start with one or more series of 4-space indents, followed by a hard tab,
 			// followed by the test output
 			// Top-level tests start with a hard tab.

--- a/tests/13-report.xml
+++ b/tests/13-report.xml
@@ -16,7 +16,7 @@
 		<properties>
 			<property name="go.version" value="1.0"></property>
 		</properties>
-		<testcase classname="failing1" name="build failed" time="0.000">
+		<testcase classname="failing1" name="[build failed]" time="0.000">
 			<failure message="Failed" type="">failing1/failing_test.go:15: undefined: x</failure>
 		</testcase>
 	</testsuite>
@@ -24,8 +24,16 @@
 		<properties>
 			<property name="go.version" value="1.0"></property>
 		</properties>
-		<testcase classname="failing2" name="build failed" time="0.000">
+		<testcase classname="failing2" name="[build failed]" time="0.000">
 			<failure message="Failed" type="">failing2/another_failing_test.go:20: undefined: y</failure>
+		</testcase>
+	</testsuite>
+	<testsuite tests="1" failures="1" time="0.000" name="package/name/setupfailing1">
+		<properties>
+			<property name="go.version" value="1.0"></property>
+		</properties>
+		<testcase classname="setupfailing1" name="[setup failed]" time="0.000">
+			<failure message="Failed" type="">setupfailing1/failing_test.go:4: cannot find package &#34;other/package&#34; in any of:&#xA;&#x9;/path/vendor (vendor tree)&#xA;&#x9;/path/go/root (from $GOROOT)&#xA;&#x9;/path/go/path (from $GOPATH)</failure>
 		</testcase>
 	</testsuite>
 </testsuites>

--- a/tests/13-syntax-error.txt
+++ b/tests/13-syntax-error.txt
@@ -2,6 +2,11 @@
 failing1/failing_test.go:15: undefined: x
 # package/name/failing2
 failing2/another_failing_test.go:20: undefined: y
+# package/name/setupfailing1
+setupfailing1/failing_test.go:4: cannot find package "other/package" in any of:
+	/path/vendor (vendor tree)
+	/path/go/root (from $GOROOT)
+	/path/go/path (from $GOPATH)
 === RUN TestA
 --- PASS: TestA (0.10 seconds)
 PASS
@@ -12,3 +17,4 @@ PASS
 ok      package/name/passing2 0.100s
 FAIL    package/name/failing1 [build failed]
 FAIL    package/name/failing2 [build failed]
+FAIL    package/name/setupfailing1 [setup failed]

